generators/app/index.js

```diff
@@ -441,20 +460,15 @@ module.exports = class extends BaseGenerator {
                 }
 
                 if (!this.options['skip-git']) {
+                    this.debug('Committing files to git');
                     this.isGitInstalled(code => {
-                        if (code === 0) {
-                            this.gitExec('rev-parse --is-inside-work-tree', { trace: false }, (err, gitDir) => {
-                                if (!gitDir) {
-                                    this.gitExec('init', { trace: false }, () => {
-                                        this.gitExec('add -A', { trace: false }, () => {
-                                            this.gitExec(
-                                                `commit -am "Initial application generated by JHipster-${this.jhipsterVersion}"`,
-                                                { trace: false },
-                                                () => this.log(chalk.green.bold('Application successfully committed to Git.'))
-                                            );
-                                        });
-                                    });
-                                }
+                        if (code === 0 && this.gitInitialized) {
+                            this.gitExec('add -A', { trace: false }, () => {
+                                this.gitExec(
+                                    `commit -am "Initial application generated by JHipster-${this.jhipsterVersion}"`,
+                                    { trace: false },
+                                    () => this.log(chalk.green.bold('Application successfully committed to Git.'))
+                                );
                             });
                         } else {
                             this.warning('The generated application could not be added to Git, as Git is not installed on your system');
```

Comment:
This message could also be changed : "The generated application could not be added to Git, as Git repository is not initiated"

Reply:
I'll update it later